device/api/umd/device/cluster.h

```diff
@@ -97,6 +98,15 @@ class tt_device {
         throw std::runtime_error("---- tt_device::configure_tlb is not implemented\n");
     }
 
+    virtual void configure_tlb(
+        chip_id_t logical_device_id,
+        tt::umd::CoreCoord core,
+        int32_t tlb_index,
+        uint64_t address,
+        uint64_t ordering = TLB_DATA::Relaxed) {
+        throw std::runtime_error("---- tt_device::configure_tlb is not implemented\n");
```

Comment:
Consider using pure virtual functions instead of runtime throw. It is easier to find places during compile time instead having 100% code coverage with tests...